### PR TITLE
Corrected Clean Arch Violations + Some more clean-up

### LINKED ISSUE #65: Reference to User from Views
We have a direct reference to `User` from some of the views, namely `HomeScreen` and `LoginScreen`. We can solve this by implementing the observer design pattern.

### PR DESCRIPTION
**Pull request by muzammil004**
# Description

Please include a summary of the changes that you have made. Also include the motivation for these changes.
- Corrected the Clean Architecture violations that were outlined in Issue #65
- ContactScreen is now uneditable
- Changed package structure of "test" to match "main"
- Got rid of a few more warnings
- Got rid of a bug that would redraw the HomeScreen every time you customized your profile. This was because we were calling HomeScreen.setState() in ProfileScreen's actionPerformed method. 
  - To solve this, I moved the lines of code that worked with drawing the HomeScreen into another method called finalizeScreen() and called finalizeScreen() only when the user presses "Login" (along with the setState() method).
- Added some logic to the entities
- Record Button on the Chat Screen is now better size

Fixes #65

# Design Patterns

Please mention which - if any - design patterns that you have implemented in this pull request.
- N/A

# SOLID Principles

Please mention which - if any - SOLID principles you have followed in this pull request.
- N/A
# Breaking Change
- [ ] Check this if your changes will cause merge conflicts in other branches.

# Checklist:
- [x] My code follows Java style conventions
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added Javadocs for new classes and methods
- [x] My changes generate no new relevant warnings
- [x] I have added tests for any new functionality added
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by erji2022 — DISMISSED** [review]
Looks good to me

**Review by terenceliu88 — APPROVED** [review]
Looks great!

File: src/main/java/message_search_use_case/MessageSearchResponse.java [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
+    private final List<Map<String, String>> messages;
 
     /**
      * Constructor for MessageSearchResponse.
      * @param text text that was searched
      * @param success whether message search was successful
      * @param e exception that was thrown if success is false, null otherwise
      */
-    public MessageSearchResponse(String text, List<Message> messages, boolean success, Exception e) {
+    public MessageSearchResponse(String text, List<Map<String, String>> messages, boolean success, Exception e) {
```
Comment (SlothfulAlfred) [inline src/main/java/message_search_use_case/MessageSearchResponse.java:18]: Missing an @param tag in the javadoc.

File: src/main/java/user_send_message/MessageInputBoundary.java [hunk 2]
```diff
@@ -20,15 +17,15 @@ public interface MessageInputBoundary {
 
     /**
      * Fetches all the messages in a specific chat ID
-     * @param chatID
+     * @param chatID ID of the chat
      * @return List of all messages in a chat
```
Comment (SlothfulAlfred) [inline src/main/java/user_send_message/MessageInputBoundary.java:21]: forgot to update the javadoc. This method does not return a list of all messages.

File: src/main/java/user_send_message/MessageInteractor.java [hunk 3]
```diff
@@ -54,23 +51,33 @@ public SendMessageResponse sendMessage(int chatID, String message, int senderID,
...
         sendMessageGateway.sendMessage(chatID, messsageToSend);
-        return new SendMessageResponse(messsageToSend, true, null);
+        return new SendMessageResponse(messsageToSend.getId(), true, null);
     }
 
     /**
      * Retrieves all the messages in a chat
-     * @param chatID
+     * @param chatID ID of the chat
      * @return List of all messages in a chat
```
Comment (SlothfulAlfred) [inline src/main/java/user_send_message/MessageInteractor.java:60]: Forgot to update @return tag in the javadoc.

**Review by SlothfulAlfred — DISMISSED** [review]
Great job!

**Review by SlothfulAlfred — APPROVED** [review]

**Review by erji2022 — APPROVED** [review]
Looks good to me!

**Review by Dismany0 — APPROVED** [review]
Mostly cleaning up a lot of files
Approved

**Pull request merged**